### DISCUSSION TITLE
Merge unique list elements from the same structs

### DISCUSSION THREAD

**Discussion by SephVelut** [cue-lang/cue#1691] [Q&A] [answered]
Given this
```cue
#Spec: {values: [...uint8]}
```

then this
```cue
impl: #Spec & {values: [1]}
impl: #Spec & {values: [1]}
```

I get back an expected `impl: { values: [1] }`.

But what if I do this instead

```cue
impl: #Spec & {values: [1]}
impl: #Spec & {values: [2]}
```

I would like back `impl: { values: [1, 2] }`. How to achieve this?

**Comment by eonpatapon** [comment, marked as answer]
This would be possible with sets or associative lists but hey are not implemented yet. See https://github.com/cue-lang/cue/issues/165#associative-lists

In the mean time you could use a temp struct to generate a list from it:

```cue
import (
	"strconv"
)

#Spec: {
	_vals: [=~"^[0-9]+$"]: true
	values: [...uint8]
	values: [ for k, _ in _vals {strconv.Atoi(k)}]
}

impl: #Spec
impl: {_vals: "1": _}
impl: {_vals: "2": _}
```

```
impl: {
    values: [1, 2]
}
```

Because struct keys are unique you'll end up with only unique elements in the list. However struct keys can only be strings.
